lib/view_component/base.rb

```diff
@@ -97,25 +99,37 @@ def render_in(view_context, &block)
       @current_template = old_current_template
     end
 
+    # Called before rendering the component.
+    #
+    # @return [void]
     def before_render
       before_render_check
     end
 
+    # Called after rendering the component.
+    #
+    # @deprecated Use `before_render` instead. Will be removed in v3.0.0.
+    # @return [void]
     def before_render_check
       # noop
     end
 
+    # Whether the ViewComponent should render.
```

Comment:
```suggestion
    # Overridable lifecycle method that determines if the component should render.
```

Similar to the above comment, thoughts on mentioning it's an overridable lifecycle method?